docs/hyperion-setup.md

```diff
@@ -1 +1,411 @@
 # Installing and configuring your first Hyperion pipeline
+
+> This document assumes that you have a basic understanding of what Hyperion is and isn't. See [What is Hyperion?](/README.md#what-is-hyperion) for an overview of Hyperion and its core features.
+
+This document will show you the core concepts behind the Hyperion pipeline and detail how to set up a simple pipeline that takes your logs from your logging source and aggregate them for later visualization in the IDE.
+
+For this tutorial, we will be using [mock-elastic-logging](https://github.com/nickyu42/mock-elastic-logging/) as the project whose logs we will aggregate. It is a simple Java project that will continuously print various dummy logs. It also contains some docker-compose setups for creating a simple ELK stack that aggregates those logs.
+
+To get started, clone the repository and follow the README to setup the ELK instance and logger. If everything is working, you should be able to open the Kibana interface at http://localhost:5601 and see the dummy logs appearing in the log tab.
+
+The final configuration files of the pipeline we will set up in this tutorial can be found in the [example](/example/) folder. If you're having issues with setup, you can reference those files for a working version.
+
+Without further ado, let's have a look at how the Hyperion pipeline works and how we can configure one ourselves.
+
+### What's the pipeline?
+
+The entire Hyperion pipeline has a single job: attach enough metadata to the incoming log lines to allow the aggregator to do it's job. Because Hyperion aims to be dynamic enough to work in a large variety of logging setups, the composable pipeline allows for any combination of plugins that resolve or attach this metadata.
+
+If we visualize the pipeline setup used in this tutorial, it'd look something like this:
+
+![](https://i.imgur.com/lRIIpBf.png)
```

Comment:
Again (perhaps we should discuss this). Is it better to host our images on the repository itself or keep it on a 3rd party website?